core/ir.ml

```diff
@@ -76,6 +76,12 @@ and binding =
 and special =
   [ `Wrong of Types.datatype
   | `Database of value
+  | `Lens of value * Types.lens_sort
+  | `LensDrop of value * string * string * value * Types.lens_sort
+  | `LensSelect of value * Types.lens_phrase * Types.lens_sort
+  | `LensJoin of value * value * string list * Types.lens_phrase * Types.lens_phrase * Types.lens_sort
```

Comment:
I am somewhat skeptical about `lens_phrase` appearing here. It is the same representation that is used in the frontend. I would expect something that has been translated to appear here.